### DISCUSSION TITLE
Using FileMeshComponent of THM and TensorMechanics together in the same input file

### DISCUSSION THREAD

**Discussion by SomnusYu** [idaholab/moose#27632] [Q&A General]
### Check these boxes if you have followed the posting rules.

- [X] Q&A General is the most appropriate section for my question
- [X] I have consulted the posting Guidelines on the Discussions front page
- [X] I have searched the Discussions forum and my question has not been asked before
- [X] I have searched the MOOSE website and the documentation does not answer my question
- [X] I have formatted my post following the posting guidelines (screenshots as a last resort, triple back quotes around pasted text)

### Question

Hi all,

I would like to use FileMeshComponent of [THM], [TensorMechanics], and [heat conduction] together in the same input file to do the fully coupled simulation. 
If the [TensorMechanics] is not included in this file, there is no problem. If I add the [TensorMechanics] part in the input file, the error shows:

![image](https://github.com/idaholab/moose/assets/118275988/c04fc918-9355-418e-a432-4dba9458439f)

Here is the simple input file:
[file_mesh_component.txt](https://github.com/idaholab/moose/files/15323318/file_mesh_component.txt)

I checked the position of this error in `QuasiStaticSolidMechanicsPhysics`:
![image](https://github.com/idaholab/moose/assets/118275988/42a46f7a-aecf-460d-b654-7bd7372ea2e4)

I think each block and subdomain is defined. Why did this error occur?

**Comment by GiudGiud** [comment]
Hello

This is not possible currently. Unless you create a SolidMechanicsComponent that does a lot of what the TensorMechanics action does. This is what was was done for the FlowComponentNS

I am reworking components in this PR
https://github.com/idaholab/moose/pull/27593
Once I m done there you ll be able to instantiate any of 'Physics' on this kind of component.

Guillaume

**Reply by SomnusYu** [reply]
I got it, thanks!
